### ISSUE TITLE
does this Patcher work on today's official release ?

### ISSUE THREAD

**Issue by WaveBird** [pookjw/SidecarPatcher#8]
does this Patcher work on today's official release? thanks

**Comment by Tobias-Bauer** [comment]
Not for me. Please reply if it worked for you

**Comment by WaveBird** [comment]
> Not for me. Please reply if it worked for you

what is the model of your mac? I am preparing the parallels Desktop env for the testing, my mbp is 2015 mid

**Comment by Tobias-Bauer** [comment]
2015 mid I also tried to connect it over USB - USBC cable but sidecar doesn't show any devices

**Comment by lucien-guoheng** [comment]
for me, it works but only in wired mode.
My devices are MBP 2016 and iPad Air 2 
but there is a little bit difference from beta version, that is in the official version, if you don't connect them via USB, you won't see any device in the status bar (but in beta version, it shows device but when you try to connect it shows error code 32002). 
In wired mode, it works perfectly, I truly hope that the new version of this tweak can fix the issue of sidecar in wireless mode.

**Comment by Tobias-Bauer** [comment]
I reinstalled Catalina but now only the wireless mode is working with Bluetooth turned of and a wired connection it won’t work

**Comment by mostafamohmed88** [comment]
yup it worked for me but ipad screen is flickering like analog Tvs

**Comment by Tobias-Bauer** [comment]
@WaveBird I installed it on a external drive and will now reinstall my internal system without a Backup(with backup didn’t work for me). The wireless mode has a bad resolution and is also flickering. Maybe [this](http://dev.zeppel.eu/luca/SidecarCorePatch) patch works.

**Comment by wdongxv** [comment]
MacBook Pro (Retina, 13-inch, Late 2013)
iPad Air 2

Using a USB cable works fine!

**Comment by Jie211** [comment]
MacBook Pro (Retina, 13-inch, Late 2013)
iPad Pro 9.7 inch
works file with USB cable

**Comment by MarshKOR** [comment]
> MacBook Pro (Retina, 13-inch, Late 2013)
> iPad Pro 9.7 inch
> works fine with or without USB cable

Is there no flickering ? And how's the graphic quality?

**Comment by w0lfschild** [comment]
> > MacBook Pro (Retina, 13-inch, Late 2013)
> > iPad Pro 9.7 inch
> > works fine with or without USB cable
> 
> Is there no flickering ? And how's the graphic quality?

If you are using a _supported Mac_ and an _unsupported iPad_ **the quality will be good** and there will be no flickering but you will **have to use a wired connection**

If you are using a _supported iPad_ and an _unsupported Mac_ **the quality will be bad** and there **may be flickering**

If you are using an _unsupported iPad_ and an _unsupported Mac_ **the quality will be bad** and there **may be flickerin**g and you will **have to use a wired connection**

**Comment by ben-z** [comment]
Didn't work for me in the official version (10.15 (19A602)) so I made my own: https://github.com/ben-z/free-sidecar

**Comment by pookjw** [comment]
> Didn't work for me in the official version (10.15 (19A602)) so I made my own: https://github.com/ben-z/free-sidecar

I read your code. You did a nice work with SwiftUI. I am learning Swift and SwiftUI these days so your code was very helpful. and seems like your patch method is similar to my method.

- My method is replace all model number to zero: `MacBookAir9,2` -> `MacBookAir0,0`

- Your method is replace selected model using `~` operator (right? What `{ $0 | 0xC0 }` means from [this line?](https://github.com/ben-z/free-sidecar/blob/010111b2e022790da6dfe235ba9734225da9ec64/free-sidecar/utils.swift#L145)): `MacBookAir9,2` -> `MacBookAir˜,2`

But you said my method didn't work for you, and your method worked. Why this happens?

**Comment by ben-z** [comment]
Hi @pookjw, thanks for checking it out! My method is masking the two most significant bits of each hex nibble (i.e. suppose we have `iPad5,4`, ascii `5` has hex value `0x35`. We change it to `0xF5`). Note that this operation is reversible.

I think the reason that the method in this repo didn't work for me is that here, the entire hex string is hard-coded. My system seems to have a slightly different hex string that the one hard coded here. So the replacement didn't find any results.
